sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample09_Extensibility.md

```diff
@@ -0,0 +1,395 @@
+Here is how we define our extension methods so that these types can be created via the `ServiceBusClient`:
+
+```C# Snippet:ServiceBusExtensions
+public static PluginSender CreatePluginSender(
+    this ServiceBusClient client,
+    string queueOrTopicName,
+    IEnumerable<Func<ServiceBusMessage, Task>> plugins)
+{
+    return new PluginSender(queueOrTopicName, client, plugins);
+}
+
+public static PluginReceiver CreatePluginReceiver(
+    this ServiceBusClient client,
+    string queueName,
+    IEnumerable<Func<ServiceBusReceivedMessage, Task>> plugins,
+    ServiceBusReceiverOptions options = default)
+{
+    return new PluginReceiver(queueName, client, plugins, options ?? new ServiceBusReceiverOptions());
+}
+
+public static PluginReceiver CreatePluginReceiver(
+    this ServiceBusClient client,
+    string topicName,
+    string subscriptionName,
+    IEnumerable<Func<ServiceBusReceivedMessage, Task>> plugins,
+    ServiceBusReceiverOptions options = default)
+{
+    return new PluginReceiver(topicName, subscriptionName, client, plugins, options ?? new ServiceBusReceiverOptions());
+}
+
+public static PluginProcessor CreatePluginProcessor(
+    this ServiceBusClient client,
+    string queueName,
+    IEnumerable<Func<ServiceBusReceivedMessage, Task>> plugins,
+    ServiceBusProcessorOptions options = default)
+{
+    return new PluginProcessor(queueName, client, plugins, options ?? new ServiceBusProcessorOptions());
+}
+
+public static PluginProcessor CreatePluginProcessor(
+    this ServiceBusClient client,
+    string topicName,
+    string subscriptionName,
+    IEnumerable<Func<ServiceBusReceivedMessage, Task>> plugins,
+    ServiceBusProcessorOptions options = default)
+{
+    return new PluginProcessor(topicName, subscriptionName, client, plugins, options ?? new ServiceBusProcessorOptions());
+}
+
+public static PluginSessionProcessor CreatePluginSessionProcessor(
+    this ServiceBusClient client,
+    string queueName,
+    IEnumerable<Func<ServiceBusReceivedMessage, Task>> plugins,
+    ServiceBusSessionProcessorOptions options = default)
+{
+    return new PluginSessionProcessor(queueName, client, plugins, options ?? new ServiceBusSessionProcessorOptions());
+}
+
+public static PluginSessionProcessor CreatePluginSessionProcessor(
+    this ServiceBusClient client,
+    string topicName,
+    string subscriptionName,
+    IEnumerable<Func<ServiceBusReceivedMessage, Task>> plugins,
+    ServiceBusSessionProcessorOptions options = default)
+{
+    return new PluginSessionProcessor(topicName, subscriptionName, client, plugins, options ?? new ServiceBusSessionProcessorOptions());
+}
+```
+
+Finally, here is how consuming code might use these types.
+Since we have derived from the library types, the only thing that would need to be updated to use this pattern is the place where the derived types are created. All other usages can be left as is:
```

Comment:
```suggestion
Finally, here is how consuming code might use these types. Since we have derived from the library types, the only thing that would need to be updated to use this pattern is the place where the derived types are created. All other usages can be left as is:
```